source/upgrading/upgrade/upgrade-4.13.rst

```diff
@@ -0,0 +1,360 @@
+   yum repository, substitute your own URL for the ones used in these examples.
+
+The first order of business will be to change the yum repository
+for each system with CloudStack packages. This means all
+management servers, and any hosts that have the KVM agent.
+
+(No changes should be necessary for hosts that are running VMware
+or Xen.)
+
+Start by opening ``/etc/yum.repos.d/cloudstack.repo`` on any
+systems that have CloudStack packages installed.
+
+This file should have content similar to the following:
+
+.. parsed-literal::
+
+   [apache-cloudstack]
+   name=Apache CloudStack
+   baseurl=http://download.cloudstack.org/centos/6/4.12/
+   enabled=1
+   gpgcheck=0
+
+If you are using the community provided package repository, change
+the base url to:
+
+.. parsed-literal::
+
+   http://download.cloudstack.org/centos/$releasever/|version|/
+
+Setup the GPG public key if you wish to enable ``gpgcheck=1``:
+
+.. parsed-literal::
+
+   rpm --import http://download.cloudstack.org/RPM-GPG-KEY
+
+
+
+If you're using your own package repository, change this line to
+read as appropriate for your |version| repository.
+
+#. Now that you have the repository configured, it's time to upgrade the
+   ``cloudstack-management``.
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-management
+
+#. If you use CloudStack usage server
+
+   .. parsed-literal::
+
+      $ sudo yum upgrade cloudstack-usage
+
+.. _upg_hyp_412:
+
+Upgrade Hypervisors
+-------------------
+
+Hypervisor: XenServer
+#####################
+
+
+No additional steps are required wrt for XenServer Hypervisor for this upgrade.
+
+
+Hypervisor: VMware
+###################
+
+.. warning::
+   For VMware hypervisor CloudStack management server packages must be
+   build using "noredist". Refer to :ref:`building-noredist`.
+
+
+No additional steps are requried for the VMware Hypervisor for this upgrade.
+
+
+.. _kvm412:
+
+Hypervisor: KVM
+#################
+
+KVM on Ubuntu
+""""""""""""""
+
+(KVM only) Additional steps are required for each KVM host. These
+steps will not affect running guests in the cloud. These steps are
+required only for clouds using KVM as hosts and only on the KVM
+hosts.
+
+#. Configure the :ref:`APT repo <apt-repo411>` as detailed above.
+
+#. Stop the running agent.
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-agent stop
+
+#. Update the agent software.
+
+   .. parsed-literal::
+
+      $ sudo apt-get upgrade cloudstack-agent
+
+#. Verify that the file ``/etc/cloudstack/agent/environment.properties`` has a
+   line that reads:
+
+   .. parsed-literal::
+
+      paths.script=/usr/share/cloudstack-common
+
+   If not, add the line.
+
+#. Start the agent.
+
+   .. parsed-literal::
+
+      $ sudo service cloudstack-agent start
+
+
+KVM on CentOS/RHEL
+"""""""""""""""""""
+
+For KVM hosts, upgrade the ``cloudstack-agent`` package
+
+#. Configure the :ref:`rpm-repo411` as detailed above.
```

Comment:
rpm-repo413

Reply:
I'm not sure how they're used, it's been from before I did not change it. Will need review/checking. It's also in the previous 4.11, 4.12 rst files. cc @PaulAngus

Reply:
fixed